CONTRIBUTING.md

```diff
@@ -1,43 +1,91 @@
+2. **Make sure of the following:**
+   * [x] There is no [opened or closed issues](https://github.com/zurb/foundation-sites/issues?utf8=%E2%9C%93&q=is%3Aissue) similar to your request
+   * [x] This is a missing feature and not a bug
+3. **Prepare a clear use case for the requested feature**.
+   This will help you to make sure that this is the feature you want, and us to better understand your needs and how it would benefits to everyone the best way.
+4. **Create your issue** ([link](https://github.com/zurb/foundation-sites/issues/new)).
+   Please provide a complete description of the feature you want, the use case you prepared will help you for that. After you created your issue, if you feel ready, you can start working on a pull request (please tell us so). See [Contributions](#contributions) below.
 
 ## Contributions
 
-All new features and bug fixes should be submitted as pull requests, so the community can review and discuss them. Core Team members can commit directly to the repository for very small changes, but should generally also submit new code as a pull request.
+All new features and bug fixes should be submitted as pull requests, so the community can review and discuss them. The rule is the same for everyone, for new contributors as for Core Team members.
+
+Before working on a bug fix or a new feature, please make sure of the following:
+* [x] **There is no similar pull request that was rejected or is not merged yet** ([link](https://github.com/zurb/foundation-sites/issues?utf8=%E2%9C%93&q=is%3Apr)).
+  Add a comment on this pull request otherwise. Explain us why this pull request is urgent or important to you.
+* [x] **There is an open issue related to this bug or feature**.
+  Please create one otherwise. It is often useful to talk about the bug fix or feature and the best way to implement it before working on it. You can skip this step for obvious changes (like typo in the documentation).
+
+After you made these checks, please follow these advices to create your pull requests:
+1. **Work on a dedicated Git branch**.
+   So you default `develop` branch stay clean and you can open multiple Pull Requests at the same time for various issues. See (2) below for the branch name format.
+2. **Use our standard format for branch, commits and pull request names**.
+   It must reference the related issue, be written in the "imperative" form (like if it was completing `now the software should...`) and be prefixed by a type (`feat` for new feature, `fix` if you repair something, `docs` documentation, `refactor` for non-breaking code cleaning, `style` for code formatting, `tests` for units or visual tests or `chore` for boring day-to-day tasks not affecting the actual code. See the [AngularJs Git Commit Message Convention](https://gist.github.com/stephenparish/9941e89d80e2bc58a153)). For example: `docs: improve Dropdown usage example #123` for commit/pull request names and `docs/dropdown-improve-usage-example-123` for the branch name.
```

Comment:
`docs` documentation => `docs` for documentation  
`tests` for units or visual => `tests` for unit (tests) or visual

Reply:
> tests for units or visual => tests for unit (tests) or visual

Why ?

Reply:
Never heard of units tests, I just know unit tests ;-)

Reply:
Ah :D

Reply:
So "or unit or visual tests"

Reply:
Right.